datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java

```diff
@@ -137,4 +150,62 @@ public BrowseResults browse(@Nonnull List<String> path,
         return BrowsePathsMapper.map(result);
     }
 
+    @Override
+    public Class<GlossaryTermUpdateInput> inputClass() {
+        return GlossaryTermUpdateInput.class;
+    }
+
+    @Override
+    public GlossaryTerm update(@Nonnull String urn, @Nonnull GlossaryTermUpdateInput input, @Nonnull QueryContext context) throws Exception {
+        if (isAuthorized(urn, input, context)) {
+            final CorpuserUrn actor = CorpuserUrn.createFromString(context.getAuthentication().getActor().toUrnStr());
+            final GlossaryTermSnapshot glossaryTermSnapshot = GlossaryTermUpdateInputSnapshotMapper.map(input, actor);
+            final Snapshot snapshot = Snapshot.create(glossaryTermSnapshot);
+
+            try {
+                Entity entity = new Entity();
+                entity.setValue(snapshot);
+                _entityClient.update(entity, context.getAuthentication());
+            } catch (RemoteInvocationException e) {
+                throw new RuntimeException(String.format("Failed to write entity with urn %s", input.getUrn()), e);
+            }
+
+            return load(urn, context).getData();
+        }
+        throw new AuthorizationException("Unauthorized to perform this action. Please contact your Data workbench support.");
+    }
+
+    private boolean isAuthorized(@Nonnull String urn, @Nonnull GlossaryTermUpdateInput update, @Nonnull QueryContext context) {
+        // Decide whether the current principal should be allowed to update the Dataset.
+        final DisjunctivePrivilegeGroup orPrivilegeGroups = getAuthorizedPrivileges(update);
+        return AuthorizationUtils.isAuthorized(
+                context.getAuthorizer(),
+                context.getAuthentication().getActor().toUrnStr(),
+                PoliciesConfig.GLOSSARY_TERM_PRIVILEGES.getResourceType(),
+                urn,
+                orPrivilegeGroups);
+    }
+
+    private DisjunctivePrivilegeGroup getAuthorizedPrivileges(final GlossaryTermUpdateInput updateInput) {
+
+        final ConjunctivePrivilegeGroup allPrivilegesGroup = new ConjunctivePrivilegeGroup(ImmutableList.of(
+                PoliciesConfig.EDIT_ENTITY_PRIVILEGE.getType()
+        ));
+
+        List<String> specificPrivileges = new ArrayList<>();
+
+        if (updateInput.getEditableSchemaMetadata() != null) {
+            specificPrivileges.add(PoliciesConfig.EDIT_GLOSSARY_TERM_COL_TAGS_PRIVILEGE.getType());
+            specificPrivileges.add(PoliciesConfig.EDIT_GLOSSARY_TERM_COL_DESCRIPTION_PRIVILEGE.getType());
```

Comment:
are we sure we want to support tagging of glossary terms schemas? This seems like it could lead to schema tagging inception, since schemas can be tagged with terms which can then have their own schemas with their own tags- is this going to get confusing?

cc @shirshanka

Reply:
removed